packages/purpose-process/test/testGetPurposes.ts

```diff
@@ -0,0 +1,485 @@
+      const result = await purposeService.getPurposes(
+        producerId1,
+        {
+          eservicesIds: [mockEService1ByTenant1.id],
+          consumersIds: [],
+          producersIds: [],
+          states: [],
+          excludeDraft: undefined,
+        },
+        { offset: 0, limit: 50 },
+        genericLogger
+      );
+      expect(result.totalCount).toBe(2);
+      expect(result.results).toEqual([mockPurpose1, mockPurpose2]);
+    });
+    it("should get the purposes if they exist (parameters: consumersIds)", async () => {
+      const result = await purposeService.getPurposes(
+        producerId1,
+        {
+          eservicesIds: [],
+          consumersIds: [consumerId1],
+          producersIds: [],
+          states: [],
+          excludeDraft: undefined,
+        },
+        { offset: 0, limit: 50 },
+        genericLogger
+      );
+      expect(result.totalCount).toBe(3);
+      expect(result.results).toEqual([
+        mockPurpose1,
+        mockPurpose5,
+        mockPurpose6,
+      ]);
+    });
+    it("should get the purposes if they exist (parameters: producersIds)", async () => {
+      const result = await purposeService.getPurposes(
+        producerId1,
+        {
+          eservicesIds: [],
+          consumersIds: [],
+          producersIds: [producerId2],
+          states: [],
+          excludeDraft: undefined,
+        },
+        { offset: 0, limit: 50 },
+        genericLogger
+      );
+      expect(result.totalCount).toBe(2);
+      expect(result.results).toEqual([mockPurpose4, mockPurpose6]);
+    });
+    it("should get the purposes if they exist (parameters: states)", async () => {
+      const result = await purposeService.getPurposes(
+        producerId1,
+        {
+          eservicesIds: [],
+          consumersIds: [],
+          producersIds: [],
+          states: [purposeVersionState.rejected, purposeVersionState.active],
+          excludeDraft: undefined,
+        },
+        { offset: 0, limit: 50 },
+        genericLogger
+      );
+      expect(result.totalCount).toBe(4);
+      expect(result.results).toEqual([
+        mockPurpose3,
+        mockPurpose4,
+        mockPurpose5,
+        mockPurpose6,
+      ]);
+    });
+    it("should get the purposes if they exist (parameters: states, archived and non-archived)", async () => {
+      const result = await purposeService.getPurposes(
+        producerId1,
+        {
+          eservicesIds: [],
+          consumersIds: [],
+          producersIds: [],
+          states: [
+            purposeVersionState.archived,
+            purposeVersionState.active,
+            purposeVersionState.rejected,
+          ],
+          excludeDraft: undefined,
+        },
+        { offset: 0, limit: 50 },
+        genericLogger
+      );
+      expect(result.totalCount).toBe(4);
+      expect(result.results).toEqual([
+        mockPurpose3,
+        mockPurpose4,
+        mockPurpose5,
+        mockPurpose6,
+      ]);
+    });
```

Comment:
@galales this is the test you suggested. One note: `mockPurpose6` has an `archived` and an `active` version. In this way it is included in the results (the filter is by `archived, active, rejected`) because it passes the `active` filter, even though it doesn't pass the `archived` filter. Is this ok? Otherwise we might have to restructure the query.

Reply:
correct, that is the expected behaviour, thanks!